### PR TITLE
Detector logic

### PR DESCRIPTION
**Pull request by rayanajakane**
Changement de logique dans detector.py

- [x] Enlever la responsabilité de filtrer le label de la fonction object_detection
- [x] Retourner un dictionnaire d'objets détectés associés à leurs coordonnées selon le temps
- [x] Ajouter le gestion de outlier dans la fonction object_detection

Notes:

- Le paramètre label est toujours utilisé dans la fonction object_detection, mais pas pour filtrer les objets. 
- Le dictionnaire d'objets utilise le raw_label comme key
- Les outliers sont filtrés par défaut à plus de 7m, mais la distance peut être modifiée

### CODE REVIEW

File: src/detector.py [hunk 1]
```diff
@@ -175,7 +175,7 @@ def object_detection(label: int, duration: int, opt):
     # Set-up Timer
     timeout = time.time() + duration
 
-    coordinated_target_list = np.empty((0,3))
+    coordinated_target_dict = dict()
```
Comment (nguyenm31) [inline src/detector.py:178]: coordinated_target_dict = {}

File: src/detector.py [hunk 2]
```diff
@@ -202,10 +202,12 @@ def object_detection(label: int, duration: int, opt):
 
             object_list = objects.object_list
             for obj in object_list:
-                if obj.raw_label != label: continue
                 if len(obj.bounding_box) == 0 : continue  
                 if np.isnan(obj.position).any(): continue
-                coordinated_target_list = np.vstack([coordinated_target_list, np.array(list(obj.position))])
+                if obj.position[2] > max_distance: continue  # Filter outliers by distance. Default is 7m
```
Comment (nguyenm31) [inline src/detector.py:207]: remove  Default is 7m

File: src/detector.py (hunk 1 above)
Comment (nguyenm31) [reply src/detector.py:178]: Change pour coordinate_dict

File: src/main.py [hunk 3]
```diff
@@ -44,8 +44,8 @@ def main():
     opt = parser.parse_args()
 
     with torch.no_grad():
-        coordinated_target_list = detector.object_detection(label, 25, opt)
-        angle = find_angle(coordinated_target_list)
+        coordinated_target_dict = detector.object_detection(label, 25, opt)
```
Comment (nguyenm31) [inline src/main.py:47]: change to coordinated_dict

File: src/main.py [hunk 4]
```diff
@@ -44,8 +44,8 @@ def main():
...
+        angle = find_angle(coordinated_target_dict[label])
```
Comment (nguyenm31) [inline src/main.py:48]: utilise la fonction .get pour éviter d'avoir un erreur si on n'a pas la clé label

**Review by nguyenm31 — APPROVED** [review]
